Sources/MCP/Base/Value.swift

```diff
@@ -400,3 +400,16 @@ extension String {
         }
     }
 }
+
+// MARK: - Decode specific type from Value
+
+extension JSONDecoder {
+    public func decode<T: Decodable>(_ type: T.Type, from value: Value) throws -> T {
+        guard let data = try? JSONEncoder().encode(value) else {
+            throw DecodingError.dataCorrupted(
+                DecodingError.Context(codingPath: [], debugDescription: "Failed to encode value"))
+        }
+
+        return try self.decode(T.self, from: data)
+    }
+}
```

Comment:
Since we're using this once in the `AnyPendingRequest` initializer, I'm inclined to either inline it there or move it to a `fileprivate` extension in `Client`.